mathics/core/atoms.py

```diff
@@ -732,7 +732,7 @@ def __new__(cls, real, imag):
             self._value = value
 
             # Cache object so we don't allocate again.
-            self._complex_numbers[value] = self
```

Comment:
The cache mechanism confuse now numbers with different precisions. Let's disable this for a while.